src/modules/tcp_fallback/tcp_fallback_select.c

```diff
@@ -76,9 +61,9 @@ void unabto_tcp_fallback_select_add_to_read_fd_set(fd_set* readFds, int* maxRead
         if (con->state != CS_IDLE) {
             unabto_tcp_fallback_state st = con->tcpFallbackConnectionState;
             unabto_tcp_fallback_connection* fbConn = &fbConns[nabto_connection_index(con)];
-            if (st > UTFS_CONNECTING && st < UTFS_CLOSED && fbConn->socket != INVALID_SOCKET) {
-                FD_SET(fbConn->socket, readFds);
-                *maxReadFd = MAX(*maxReadFd ,(int)(fbConn->socket));
+            if (st > UTFS_CONNECTING && st < UTFS_CLOSED && fbConn->socket.socket != INVALID_SOCKET) {
+                FD_SET(fbConn->socket.socket, readFds);
```

Comment:
could we make a similar function for the fbConn->socket type? it's kind of implementation specific that a socket struct has a socket variable.